pkg/cmd/factory/factory_impl.go

```diff
@@ -95,10 +101,10 @@ func (f *factoryImpl) EtcdClient() (*kv.CDCEtcdClient, error) {
 		TLS:         tlsConfig,
 		LogConfig:   &logConfig,
 		DialTimeout: 30 * time.Second,
-		// TODO(hi-rustin): add gRPC metrics to Options.
-		// See also: https://github.com/pingcap/ticdc/pull/2341#discussion_r673018537.
 		DialOptions: []grpc.DialOption{
 			grpcTLSOption,
+			grpc.WithUnaryInterceptor(kv.GRPCMetrics.UnaryClientInterceptor()),
+			grpc.WithStreamInterceptor(kv.GRPCMetrics.StreamClientInterceptor()),
```

Comment:
These metrics are initialized in command line tool, which has nothing to do with cdc server, will they be pulled by the prometheus server?

Reply:
I think 'cdc server' also use the pd client.

Reply:
Yes, `cdc server` does use pd client, but this factory is not used in `cdc server`.

Reply:
Yes, it is not used in the server. But I understand they will use the same metric and it will be pulled as long as it is registered to the same registry? I'm not sure, I need to try it.

Reply:
I read the server code carefully and we should be able to use the construction factory on the server as well, with some minor changes.